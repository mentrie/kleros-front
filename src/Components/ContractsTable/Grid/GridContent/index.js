import React, { Component } from 'react'
import _ from 'lodash'
import { connect } from 'react-redux'
import { withRouter, Link } from 'react-router-dom'
import 'babel-polyfill'
import FontAwesome from 'react-fontawesome'
import { getContracts } from '../../../../redux/contracts/action-creators'
import { truncateText } from '../../../../helpers/truncateText'
import './GridContent.css'

export class GridContent extends Component {
  componentWillMount () {
    this.props.getDataContracts()
  }

  render () {
    const { hasErrored, isFetching, contracts = [] } = this.props
    if (hasErrored) {
      return <p>Sorry! There was an error loading the balance</p>
    }

    if (isFetching) {
      return (
        <div className='GridContent-container'>
          <div className='items loader'>
            <div className='linear-background-100' />
            <div className='linear-background-90' />
            <div className='linear-background-90' />
          </div>
        </div>
      )
    }

    return (
      <div className='GridContent-container'>
        <div className='items'>
          {
            _.isEmpty(contracts) &&
            <div className='items-row'>
              <div className='item item-no-contracts'>
                You have no contracts.
              </div>
            </div>
          }
          {
            contracts.filter(contract => {
              return contract.arbitrator === process.env.REACT_APP_ARBITRATOR_ADDRESS
            }).map(contract =>
              (
                <Link key={contract.address} to={`contract-summary/${contract.address}`}>
                  <div className='items-row'>
                    <div className='item-contract-hash'>{contract.address}</div>
                    <div className='item item-party-a'>{truncateText(contract.partyA, 10)}</div>
                    <div className='item item-party-b'>{truncateText(contract.partyB, 10)}</div>
                    <div className='item item-rule'>
                      <FontAwesome name='circle-thin' />
                    </div>
                  </div>
                </Link>
              )
            )
          }
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    contracts: state.contracts.contracts,
    hasErrored: state.contracts.failureDisputes,
    isFetching: state.contracts.requestDisputes
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getDataContracts: () => dispatch(getContracts())
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GridContent))
