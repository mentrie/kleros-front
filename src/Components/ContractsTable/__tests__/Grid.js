import React from 'react'
import Grid from '../Grid'

describe('Grid Exists', () => {
  test('should render grid component properly', () => {
    const component = shallow(<Grid />)
    expect(component).toMatchSnapshot()
  })
})
