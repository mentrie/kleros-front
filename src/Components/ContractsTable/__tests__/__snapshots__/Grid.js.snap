// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Exists should render grid component properly 1`] = `
<div
  className="Grid-container undefined"
>
  <div
    className="grid"
  >
    <div
      className="header"
    >
      <div
        className="item header-contract-hash"
      >
        Address
      </div>
      <div
        className="item header-party-a"
      >
        Address Party A
      </div>
      <div
        className="item header-party-b"
      >
        Address Party B
      </div>
      <div
        className="item header-rule"
      >
        Rule
      </div>
    </div>
  </div>
</div>
`;
